backend-shared/config/db/util/utils_test.go

```diff
@@ -1,12 +1,321 @@
+				Application_id:                        application.Application_id,
+				DeploymentName:                        "my-depl-to-app-mapping",
+				DeploymentNamespace:                   "my-namespace",
+				NamespaceUID:                          "test-1",
+			}
+		})
+
+		AfterEach(func() {
+			// ----------------------------------------------------------------------------
+			By("Delete resources and clean db entries created by test.")
+			// ----------------------------------------------------------------------------
+
+			_, err = dbQueries.DeleteDeploymentToApplicationMappingByDeplId(ctx, deploymentToApplicationMapping.Deploymenttoapplicationmapping_uid_id)
+			Expect(err).To(BeNil())
+
+			_, err = dbQueries.DeleteApplicationById(ctx, application.Application_id)
+			Expect(err).To(BeNil())
+
+			_, err = dbQueries.DeleteGitopsEngineInstanceById(ctx, gitopsEngineInstance.Gitopsengineinstance_id)
+			Expect(err).To(BeNil())
+
+			_, err = dbQueries.DeleteGitopsEngineClusterById(ctx, gitopsEngineCluster.Gitopsenginecluster_id)
+			Expect(err).To(BeNil())
+
+			_, err = dbQueries.DeleteManagedEnvironmentById(ctx, managedEnvironment.Managedenvironment_id)
+			Expect(err).To(BeNil())
+
+			_, err = dbQueries.DeleteClusterCredentialsById(ctx, clusterCredentials.Clustercredentials_cred_id)
+			Expect(err).To(BeNil())
+
+			// Close connection
+			dbQueries.CloseDatabase()
+		})
+
+		It("Should create new DeploymentToApplicationMapping and if called second time, it should return existing resource instead of creating new.", func() {
+
+			// ----------------------------------------------------------------------------
+			By("Create new DeploymentToApplicationMapping resource.")
+			// ----------------------------------------------------------------------------
+			isNewFirst, errFirst := GetOrCreateDeploymentToApplicationMapping(ctx, deploymentToApplicationMapping, dbQueries, log)
+			Expect(errFirst).To(BeNil())
+			Expect(isNewFirst).To(BeTrue())
+
+			// ----------------------------------------------------------------------------
+			By("Check if DeploymentToApplicationMapping resource entry is created in DB.")
+			// ----------------------------------------------------------------------------
+
+			err = dbQueries.GetDeploymentToApplicationMappingByDeplId(ctx, &db.DeploymentToApplicationMapping{
+				Deploymenttoapplicationmapping_uid_id: deploymentToApplicationMapping.Deploymenttoapplicationmapping_uid_id,
+			})
+			Expect(err).To(BeNil())
+
+			// ----------------------------------------------------------------------------
+			By("Try to create same DeploymentToApplicationMapping, it should return existing resource.")
+			// ----------------------------------------------------------------------------
+
+			isNewSecond, errSecond := GetOrCreateDeploymentToApplicationMapping(ctx, deploymentToApplicationMapping, dbQueries, log)
+			Expect(errSecond).To(BeNil())
+			Expect(isNewSecond).To(BeFalse())
```

Comment:
Similarly here (as mentioned in above comment)^^

Reply:
Yes, it could be done, my thinking here was that it Is is not repeated anywhere, just used at one place. If needed I am Okay to move it into a function. WDYT?